app/models/snapshot.rb

```diff
@@ -86,7 +86,7 @@ def self.remove_unused_evm_snapshots(delay)
   end
 
   def recently_created?
-    @recent_threshold ||= (VMDB::Config.new("vmdb").config.fetch_path(:ems_refresh, :raise_vm_snapshot_complete_if_created_within).to_i_with_method || 15.minutes)
+    @recent_threshold ||= ::Settings.ems_refresh.raise_vm_snapshot_complete_if_created_within.to_i_with_method || 15.minutes
```

Comment:
This is a bug in the original code.  `nil.to_i_with_method # => 0`, would never let the RHS get executed.